quaint/src/ast/values.rs

```diff
@@ -34,11 +35,464 @@ where
+
+    /// Creates a new datetime value.
+    pub fn datetime(value: DateTime<Utc>) -> Self {
+        ValueType::datetime(value).into_value()
+    }
+
+    /// Creates a new date value.
+    pub fn date(value: NaiveDate) -> Self {
+        ValueType::date(value).into_value()
+    }
+
+    /// Creates a new time value.
+    pub fn time(value: NaiveTime) -> Self {
+        ValueType::time(value).into_value()
+    }
+
+    /// Creates a new JSON value.
+    pub fn json(value: serde_json::Value) -> Self {
+        ValueType::json(value).into_value()
+    }
+
+    /// Creates a new XML value.
+    pub fn xml<T>(value: T) -> Self
+    where
+        T: Into<Cow<'a, str>>,
+    {
+        ValueType::xml(value).into_value()
+    }
+
+    /// `true` if the `Value` is null.
+    pub fn is_null(&self) -> bool {
+        self.typed.is_null()
+    }
+
+    /// Returns a &str if the value is text, otherwise `None`.
+    pub fn as_str(&self) -> Option<&str> {
+        self.typed.as_str()
+    }
+
+    /// `true` if the `Value` is text.
+    pub fn is_text(&self) -> bool {
+        self.typed.is_text()
+    }
+
+    /// Returns a char if the value is a char, otherwise `None`.
+    pub fn as_char(&self) -> Option<char> {
+        self.typed.as_char()
+    }
+
+    /// Returns a cloned String if the value is text, otherwise `None`.
+    pub fn to_string(&self) -> Option<String> {
+        self.typed.to_string()
+    }
+
+    /// Transforms the `Value` to a `String` if it's text,
+    /// otherwise `None`.
+    pub fn into_string(self) -> Option<String> {
+        self.typed.into_string()
+    }
+
+    /// Returns whether this value is the `Bytes` variant.
+    pub fn is_bytes(&self) -> bool {
+        self.typed.is_bytes()
+    }
+
+    /// Returns a bytes slice if the value is text or a byte slice, otherwise `None`.
+    pub fn as_bytes(&self) -> Option<&[u8]> {
+        self.typed.as_bytes()
+    }
+
+    /// Returns a cloned `Vec<u8>` if the value is text or a byte slice, otherwise `None`.
+    pub fn to_bytes(&self) -> Option<Vec<u8>> {
+        self.typed.to_bytes()
+    }
+
+    /// `true` if the `Value` is a 32-bit signed integer.
+    pub fn is_i32(&self) -> bool {
+        self.typed.is_i32()
+    }
+
+    /// `true` if the `Value` is a 64-bit signed integer.
+    pub fn is_i64(&self) -> bool {
+        self.typed.is_i64()
+    }
+
+    /// `true` if the `Value` is a signed integer.
+    pub fn is_integer(&self) -> bool {
+        self.typed.is_integer()
+    }
+
+    /// Returns an `i64` if the value is a 64-bit signed integer, otherwise `None`.
+    pub fn as_i64(&self) -> Option<i64> {
+        self.typed.as_i64()
+    }
+
+    /// Returns an `i32` if the value is a 32-bit signed integer, otherwise `None`.
+    pub fn as_i32(&self) -> Option<i32> {
+        self.typed.as_i32()
+    }
+
+    /// Returns an `i64` if the value is a signed integer, otherwise `None`.
+    pub fn as_integer(&self) -> Option<i64> {
+        self.typed.as_integer()
+    }
```

Comment:
Similarly to above, `integer` is redundant compared to `i32`.

Reply:
No, the implementation in "ValueType" differs from that of i32.